docs/wiki/99 - Developers/Creating a Module.md

```diff
@@ -0,0 +1,169 @@
+of LORIS modules are "Menu Filters" and "Forms".
+
+#### Menu Filters
+
+This is a type of module that displays a table of data with options to filter the values of this table.
+
+The correct way to create a menu filter is to include additional classes that represent the idea of a
+row in the table displayed to a user. This differs from a row in the database itself; for example, a row in the 
+menu filter module may represent a joining of columns across many database tables.
+
+If you are creating a Menu Filter module, your class `php/my_new_module.class.inc` should extend
+the class `\DataFrameworkMenu`.
+
+[See this file for an example](../../../modules/dicom_archive/php/dicom_archive.class.inc).
+
+##### Data Instance
+You should create a class `php/mynewmodulerow.class.inc` to represent a Menu Filter row. This will
+provide basic structure for the row and include permission flags that can restrict data to users
+affiliated with a given Site or Project.
+
+
+[See this file for an example](../../../modules/dicom_archive/php/dicomarchiverow.class.inc).
+
+##### Row Provisioner.
+This file should be accompanied by a class `php/mynewmodulerowprovisioner.class.inc`. This file should
+contain the SQL statement used to query the database for data that will be consolidated into a
+Menu Filter row.
+
+[See this file for an example](../../../modules/dicom_archive/php/dicomarchiverowprovisioner.class.inc).
+
+#### Forms
+
+This type of module represents an HTML form and is used to submit data to the back-end.
+
+### JavaScript files
+
+Modern LORIS modules use ReactJS to present data. The content of these files will vary significantly
+depending on what functionality you want to provide. The best way to learn about creating JavaScript
+for a new module is by studying the code of an existing module that is similar to what you are
+trying to design.
+
+Be sure to familiarize yourself with our [ReactJS guidelines](./ReactJS Guidelines.md).
+
+### SQL Files
+
+New modules must be inserted into the database. You must create new permissions for the module and insert
+an entry for your new module within the `modules` table in LORIS.
+
+#### Module permissions
+
+You must define new permissions for the module. These are used to limit users to specific actions with the module.
+
+Permissions are defined in the `permissions` table in LORIS and should be separated into view and edit permissions in most cases.
+
+In our example, you would create two new permissions: `my_new_module_view` and `my_new_module_edit`. Your PHP code within
+`php/my_new_module.class.inc` will define what these permissions allow a user to do using the `hasAccess()` function. 
+
+To add new permissions, append them to the list of existing permissions found in `SQL/0000-00-01-Permission.sql`. (This file is found in your LORIS root directory, not within the module subdirectory.)
+
+#### Modules table
+
+All modules must be listed in the file `SQL/0000-00-02-Modules.sql`. 
```

Comment:
Should add a link to the Database schema modeling standard. https://github.com/aces/Loris/blob/master/docs/SQLModelingStandard.md

Reply:
Added on line 98